### PR TITLE
Feature/login-signup-testing-and-documentation

### LINKED ISSUE #34: login/signup use case testing + documentation

### PR DESCRIPTION
**Pull request by RqedK**
This is implementation of my testing for the login and signup use cases.
Implementation of Javadoc in all the login/signup files and in the data_access files.

This should be merged to main after fiona's pull request for the branch feature/recipe to grocery list because the other changes present in this file are reflected in that branch so I didn't mention the other changes. I had branched this off of that branch.

### CODE REVIEW

**Comment by parkhangorodsky** [comment]
link an issue

**Comment by RqedK** [comment]
@parkhangorodsky Linked the issue.

**Review by parkhangorodsky — CHANGES_REQUESTED** [review]
1) write a better/more descriptive title please
2) there are changes here beyond just adding testing/documentation - add a description/overview of these to your description please 

otherwise, the testing part looks fine. ill re look once theres more info up

**Review by parkhangorodsky — APPROVED** [review]

File: .idea/encodings.xml
Comment (hhpfiona) [inline .idea/encodings.xml:1]: dont need t push

File: .idea/misc.xml
Comment (hhpfiona) [inline .idea/misc.xml:1]: dont need t push

File: src/main/java/app/Config.java
Comment (hhpfiona) [inline src/main/java/app/Config.java:1]: i don't think this is part of login testing and documentation but rather fixing the logged in user data issue no?

File: src/main/java/use_cases/_common/authentication/AuthenticationInterface.java
Comment (hhpfiona) [inline src/main/java/use_cases/_common/authentication/AuthenticationInterface.java:1]: this has some overlap with my branch - but we can deal with it when merging

File: src/main/java/use_cases/recipe_to_grocery/gui/RecipeToGroceryView.java
Comment (hhpfiona) [inline src/main/java/use_cases/recipe_to_grocery/gui/RecipeToGroceryView.java:1]: i think this is also overlap with my files and my pushes

**Review by hhpfiona — APPROVED** [review]
looks detailed in terms of documentaiton, some pushing and pulling is not super robust in terms of git, but good enough

**Review by seonghyunban — APPROVED** [review]
Next time, please make a separate branch for testing, documentation, and miscellaneous changes. Other than that, It should be fine.

**Review by hhpfiona — APPROVED** [review]

**Pull request merged**